src/components/Signup.js

```diff
@@ -1,61 +1,154 @@
 import React, { useState } from "react";
 import { connect } from "react-redux";
-import Loader from "react-loader-spinner";
+import styled, { keyframes } from "styled-components";
+import { Link } from "react-router-dom";
 import { signup } from "../actions";
 
 const Signup = props => {
-
+  console.log(props);
   const [userData, setUser] = useState({
     username: "",
     email: "",
     password: ""
   });
 
-
   const userSignup = e => {
     e.preventDefault();
 
-    props.signup(userData)
+    props.signup(userData);
   };
 
   return (
-    <div>
-      <form onSubmit={userSignup}>
-        <input
-          type="text"
-          placeholder="Enter name"
-          value={userData.username}
-          onChange={e => setUser({ ...userData, username: e.target.value })}
-        />
-        <input
-          type="email"
-          placeholder="Enter email"
-          value={userData.email}
-          onChange={e => setUser({ ...userData, email: e.target.value })}
-        />
-        <input
-          type="password"
-          placeholder="Enter password"
-          value={userData.password}
-          onChange={e => setUser({ ...userData, password: e.target.value })}
-        />
-        <button type="submit">
-          {props.signingUp ? (
-            <Loader type="ThreeDots" color="#ccc" height={80} width={80} />
-          ) : (
-            "Sign Up"
-          )}
-        </button>
-      </form>
-    </div>
+    <SignupModal display={localStorage.getItem("token") ? "none" : "block"}>
```

Comment:
What gets displayed if the token is in local storage and I clicked singup? As I see it it would just produce an empty page?

Reply:
Thanks for catching that. I have corrected it in the latest commit